src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,39 @@
+import { FC, useState, useEffect } from 'react';
+import cn from 'classnames';
+
+import { getUser } from '../../api';
+
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface Props {
+  todo: Todo;
+  onClose: () => void;
+}
+
+export const TodoModal: FC<Props> = ({ todo, onClose }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { id, title, completed, userId } = todo;
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  useEffect(() => {
+    getUser(userId)
+      .then(setUser)
+      .then(() => setIsLoading(status => !status));
+  }, [userId]);
```

Comment:
The effect hook will re-run whenever `userId` changes. If the `userId` is constant, you can pass an empty array `[]` as the second argument to `useEffect` to prevent unnecessary re-runs.